vnext/Scripts/msbuildeval.ps1

```diff
@@ -0,0 +1,73 @@
+# Evaluate an MSBuild property name
+param(
+    [Parameter(Mandatory=$true)]
+    [String]$SolutionFile,
+    [Parameter(Mandatory=$true)]
+    [String]$ProjectFile,
+    [Parameter(Mandatory=$true)]
+    [String]$PropertyName
+)
+
+function Get-MSBuildPath {
+    $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (!(Test-Path $vsWhere)) {
+        throw "Unable to find vswhere.exe."
+    }
+    $vsPath = & $vsWhere -version 16.5 -property installationPath;
+    return "$vsPath\MSBuild\Current\Bin\";
+}
+
+function Get-MSBuildProperty {
+    param (
+        [Parameter(Mandatory=$true)]
+        [String]$SolutionPath,
+        [Parameter(Mandatory=$true)]
+        [String]$ProjectPath,
+        [Parameter(Mandatory=$true)]
+        [String]$PropertyName
+    )
+
+    # Identify location of MSBuild
+    $MsBuildPath = Get-MSBuildPath
+
+    # Load Microsoft.Build.dll into script
+    Add-Type -Path "$MsBuildPath\Microsoft.Build.dll" | Out-Null
+
+    # Build a temporary "metaproj" of the solution file so it can be processed
+    ${env:MSBUILDEMITSOLUTION} = 1
+    & $MsBuildPath\MSBuild.exe $SolutionPath | Out-Null
+
+    # Build a local project collection
+    $projectCollection =[Microsoft.Build.Evaluation.ProjectCollection]::new()
+
+    # Process solution
+    $solution = [Microsoft.Build.Evaluation.Project]::new("$SolutionPath.metaproj", $null, "Current", $projectCollection)
+
+    # Clean up "metaproj" files
+    Remove-Item "$SolutionPath.metaproj" | Out-Null
```

Comment:
Why is this needed?

Reply:
Consider wrapping this in try/finally style block so that we still do it if the above line fails.

Reply:
Because if you call msbuild on just a project file, then we don't get any solution variables like `SolutionFile`, which means any evaluations that use those values will be wrong.